google/cloud/pubsub/samples/samples.cc

```diff
@@ -991,6 +991,62 @@ void SubscriberFlowControlSettings(
   (std::move(subscriber), std::move(subscription));
 }
 
+void SubscriberRetrySettings(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::testing_util;
+  if (argv.size() != 2) {
+    throw examples::Usage{
+        "subscriber-retry-settings <project-id> <subscription-id>"};
+  }
+
+  //! [subscriber-retry-settings]
+  namespace pubsub = google::cloud::pubsub;
+  using google::cloud::future;
+  using google::cloud::StatusOr;
+  [](std::string project_id, std::string subscription_id) {
+    // By default a subscriber will retry for 60 seconds, with an initial
+    // backoff of 100ms, a maximum backoff of 60 seconds, and the backoff will
+    // grow by 30% after each attempt. This changes those defaults.
+    auto subscriber = pubsub::Subscriber(pubsub::MakeSubscriberConnection(
+        pubsub::ConnectionOptions{},
+        pubsub::LimitedTimeRetryPolicy(
+            /*maximum_duration=*/std::chrono::minutes(1))
+            .clone(),
+        pubsub::ExponentialBackoffPolicy(
+            /*initial_delay=*/std::chrono::milliseconds(200),
+            /*maximum_delay=*/std::chrono::seconds(10),
+            /*scaling=*/2.0)
+            .clone()));
+    auto subscription =
+        pubsub::Subscription(std::move(project_id), std::move(subscription_id));
+
+    std::mutex mu;
+    std::condition_variable cv;
+    int count = 0;
+    auto constexpr kExpectedMessageCount = 1;
+    auto handler = [&](pubsub::Message const& m, pubsub::AckHandler h) {
+      std::move(h).ack();
+      {
+        std::lock_guard<std::mutex> lk(mu);
+        std::cout << "Received message " << m << "\n";
```

Comment:
Opt: Perhaps move the logging outside the lock?

Reply:
That was intentional:

- The output of the example is unreadable otherwise (in a larger application one would need to do something different, of course, but output from inside the lock works here).
- The performance impact is not an issue for an example methinks.

Reply:
I wasn't thinking about the performance, but rather it being an example.  But I guess the output really isn't part of the example.  In any case, ack on the serialization.